x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/topn/WKBTopNEncoder.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.compute.operator.topn;
+
+import org.apache.lucene.util.BytesRef;
+import org.elasticsearch.common.lucene.BytesRefs;
+import org.elasticsearch.compute.operator.BreakingBytesRefBuilder;
+import org.elasticsearch.geometry.Geometry;
+import org.elasticsearch.geometry.utils.GeometryValidator;
+import org.elasticsearch.geometry.utils.WellKnownBinary;
+import org.elasticsearch.geometry.utils.WellKnownText;
+
+import java.nio.ByteOrder;
+
+/**
+ * Encodes WKB bytes[].
+ * Since WKB can contain bytes with zero value, which are used as terminator bytes, we need to encode differently.
+ * Our initial implementation is to re-write to WKT and encode with UTF8TopNEncoder.
+ * This is likely very inefficient.
+ * We cannot use the UTF8TopNEncoder as is, because it removes the continuation byte, which could be a valid value in WKB.
```

Comment:
Ooof. Ouch.

Reply:
I was thinking a nicer option would be to base64 encode the binary. Might be more space efficient than converting WKB to WKT and back.

Reply:
That's terrible that we cannot support generic binary data because we are giving special meaning so some bytes.

Reply:
we should runlen prefix rows and columns. It is more efficient for parsing as we don't need to check every byte to be a special one and it provides support for generic binary data. The overhead it adds in size should be pretty small.

Reply:
ok, not sure what you want to use the utf8 encoder when we are not utf8. 

 I think geometries are not sortable so this can be implemented as non sortable and we can encode it / decode it using a runlen prefix (saving how many bytes are to be written / we need to read ).

Reply:
++. We can just runlen prefix it.

If we need to sort the geometries one day we'd come up with an encoding.

Reply:
We probably should just call this `UnsortableViableLengthBytesEncoder` or something.